File: java/flowcamel-core/src/main/java/com/flowcamel/core/model/BlockDefinition.java
package com.flowcamel.core.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.ArrayList;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class BlockDefinition {
  public String type;
  public String label;
  @JsonProperty("short")
  public String shortDescription;
  public BlockCategory category;
  public String icon;
  public String glyph;
  public String explain;
  public String camelComponent;
  public String scheme;
  public String camelUri;
  public String description;
  public List<PropSchema> props = new ArrayList<>();
}

File: packages/core/src/model/BlockDefinition.ts
import { BlockCategory } from './BlockCategory.js';

export interface PropOption {
  value: string;
  title: string;
  sub?: string;
}

export interface PropSchema {
  key: string;
  label: string;
  /** Handover baseline: text | number | password | select; catalog also uses chips, radio, textarea */
  type: 'text' | 'number' | 'password' | 'select' | 'chips' | 'radio' | 'textarea';
  placeholder?: string;
  required: boolean;
  options?: string[] | PropOption[];
  defaultValue?: string | number;
  /** Plain-English wizard question shown above the field */
  q?: string;
  /** Helper sentence below the question */
  help?: string;
  /** When true, block wizard/config can bind this field to Application properties. */
  linkable?: boolean;
}

export interface BlockDefinition {
  type: string;
  label: string;
  /** One-liner shown under block icon on sidebar */
  short: string;
  category: BlockCategory;
  icon: string;
  /** Which abstract EIP glyph to show in the wizard panel */
  glyph: 'source' | 'filter' | 'transform' | 'split' | 'log' | 'router' | 'destination';
  /** Plain-English explanation shown on step 1 of the wizard */
  explain: string;
  camelComponent: string;
  /** Camel catalog scheme (e.g. kafka, ftp). Omitted for pure EIP actions. */
  scheme?: string;
  camelUri: string;
  description: string;
  props: PropSchema[];
}
